File: app/src/main/java/com/droidtitan/volleyexamples/rest/util/OkHttpStack.java
package com.droidtitan.volleyexamples.rest.util;

import com.android.volley.toolbox.HurlStack;
import com.squareup.okhttp.OkHttpClient;
import com.squareup.okhttp.OkUrlFactory;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;

public class OkHttpStack extends HurlStack {
    private final OkUrlFactory okUrlFactory;

    public OkHttpStack() {
        this(new OkUrlFactory(new OkHttpClient()));
    }

    public OkHttpStack(OkUrlFactory okUrlFactory) {
        if (okUrlFactory == null) {
            throw new NullPointerException("Client must not be null.");
        }
        this.okUrlFactory = okUrlFactory;
    }

    @Override
    protected HttpURLConnection createConnection(URL url) throws IOException {
        return okUrlFactory.open(url);
    }
}

File: app/src/main/java/com/droidtitan/volleyexamples/rest/util/OkHttpStack.kt
package com.droidtitan.volleyexamples.rest.util

import com.android.volley.toolbox.HurlStack
import com.squareup.okhttp.OkUrlFactory
import java.io.IOException
import java.net.HttpURLConnection
import java.net.URL

public class OkHttpStack(val okUrlFactory: OkUrlFactory) : HurlStack() {
    throws(IOException::class)
    override fun createConnection(url: URL): HttpURLConnection {
        return okUrlFactory.open(url)
    }
}
